packages/git/src/browser/git-base-widget.ts

```diff
@@ -53,4 +110,65 @@ export class GitBaseWidget extends VirtualWidget {
         }
         return result;
     }
+
+    protected onAfterAttach(msg: Message): void {
+        super.onAfterAttach(msg);
+        this.addKeyListener(this.node, Key.ARROW_LEFT, () => this.handleLeft());
+        this.addKeyListener(this.node, Key.ARROW_RIGHT, () => this.handleRight());
+        this.addKeyListener(this.node, Key.ARROW_UP, () => this.handleUp());
+        this.addKeyListener(this.node, Key.ARROW_DOWN, () => this.handleDown());
+        this.addKeyListener(this.node, Key.ENTER, () => this.handleEnter());
+    }
+
+    protected handleLeft(): void {
+        this.selectPreviousNode();
+    }
+
+    protected handleRight(): void {
+        this.selectNextNode();
+    }
+
+    protected handleUp(): void {
+        this.selectPreviousNode();
+    }
+
+    protected handleDown(): void {
+        this.selectNextNode();
+    }
+
+    protected handleEnter(): void {
+
+    }
+
+    protected getSelected(): T | undefined {
+        return this.gitNodes ? this.gitNodes.find(c => c.selected || false) : undefined;
+    }
+
+    protected selectNode(node: T) {
+        const n = this.getSelected();
+        if (n) {
+            n.selected = false;
+        }
+        this.selectionService.selection = node;
+    }
+
+    protected selectNextNode() {
+        const idx = this.indexOfSelected;
+        if (idx !== undefined && idx >= 0 && idx < this.gitNodes.length - 1) {
+            this.selectNode(this.gitNodes[idx + 1]);
+        }
+    }
+
+    protected selectPreviousNode() {
+        const idx = this.indexOfSelected;
+        if (idx && idx > 0) {
+            this.selectNode(this.gitNodes[idx - 1]);
+        }
+    }
+
+    protected get indexOfSelected() {
```

Comment:
please declare return types, don't infer them, use inferred types for local variables